File: exercicios/python/Instruções Condicionais/Q12.py
mes = int(input("Digite o mês (1-12): "))
ano = int(input("Digite o ano: "))
if mes in [1,3,5,7,8,10,12]:
    print("31 dias")
elif mes in [4,6,9,11]:
    print("30 dias")
elif mes == 2:
    if (ano%400==0) or (ano%4==0 and ano%100!=0):
        print("29 dias")
    else:
        print("28 dias")
else:
    print("Mês inválido")

File: cpp/Instruções Condicionais/Q12.cpp
/*
------------------------------------------------
12. Crie um algoritmo/fluxograma/programa que 
solicite ao usuário um mês e um ano e imprima se
o mês indicado tem 28, 29, 30 ou 31 dias.
 */
#include <iostream>
#include <locale.h>
using namespace std;
int main () {
	setlocale(LC_ALL,"Portuguese");
	int ano, mes;
	cout<<"Informe o ano:";
	cin>>ano;
	cout<<"Informe o mês [1..12]:";
	cin>>mes;

	switch(mes) {
		case 1: case 3:	case 5: case 7:
		case 8: case 10: case 12:
			cout<<"O mês tem 31 dias.";
			break;
		case 2: case 4:	case 6:	case 9: case 11:
			if (mes == 2) {
				int resto = ano % 4;
				if (resto == 0) {
					cout<<"O mês tem 29 dias.";
				} else {
					cout<<"O mês tem 28 dias.";
				}
			} else {
				cout<<"O mês tem 30 dias.";
			}
			break;
		default:
			cout<<"Mês inválido!"<<endl;
	}
	return 0;
}
